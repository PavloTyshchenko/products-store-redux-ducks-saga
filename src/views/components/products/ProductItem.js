import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'react-bootstrap';

const ProductItem = ({ product: { img, name, price, currency, link } }) => {

    return (
        <Card data-test="productItemCard">
            <Card.Img variant="top" src={img} />
            <Card.Body>
                <Card.Title>
                    <a href={link}>{name}</a>
                </Card.Title>
                <Card.Text>
                    Price: {price} ({currency})
                </Card.Text>
            </Card.Body>
        </Card>
    );
};

ProductItem.propTypes = {
    product: PropTypes.object.isRequired
};

export default ProductItem;
