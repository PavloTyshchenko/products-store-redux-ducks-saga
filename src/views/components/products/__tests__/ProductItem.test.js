import React from 'react';
import { shallow } from 'enzyme';

import ProductItem from '../ProductItem';

const defaultProps = {
    product: {
        img: "test",
        name: "test",
        price: 16,
        currency: "gbp",
        link: "test"
    }
}

const setup = (propsOption) => {
    const props = Object.assign(defaultProps, propsOption);
    const wrapper = shallow(<ProductItem {...props} />);
    return {
        wrapper,
        props
    };
};

describe('ProductItem Component', () => {

    it('Should render without errors', () => {
        const { wrapper } = setup();
        const component = wrapper.find(`[data-test='productItemCard']`);
        expect(component.length).toBe(1);
    });

    test('Should match snapshot', () => {
        const { wrapper } = setup();
        expect(wrapper.debug()).toMatchSnapshot();
    });
});