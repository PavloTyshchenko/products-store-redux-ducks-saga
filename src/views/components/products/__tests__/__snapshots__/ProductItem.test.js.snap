// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem Component Should match snapshot 1`] = `
"<Card data-test=\\"productItemCard\\" body={false}>
  <CardImg variant=\\"top\\" src=\\"test\\" />
  <CardBody>
    <CardTitle>
      <a href=\\"test\\">
        test
      </a>
    </CardTitle>
    <CardText>
      Price: 
      16
       (
      gbp
      )
    </CardText>
  </CardBody>
</Card>"
`;
